pkg/clients/s3/bucketclients/notificationConfig.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bucketclients
+
+import (
+	"context"
+
+	awss3 "github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	"github.com/crossplane/provider-aws/apis/s3/v1beta1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+	"github.com/crossplane/provider-aws/pkg/clients/s3"
+)
+
+// NotificationConfigurationClient is the client for API methods and reconciling the LifecycleConfiguration
+type NotificationConfigurationClient struct {
+	config *v1beta1.NotificationConfiguration
+	client s3.BucketClient
+}
+
+// NewNotificationConfigurationClient creates the client for Accelerate Configuration
+func NewNotificationConfigurationClient(bucket *v1beta1.Bucket, client s3.BucketClient) *NotificationConfigurationClient {
+	return &NotificationConfigurationClient{config: bucket.Spec.Parameters.NotificationConfiguration, client: client}
+}
+
+func notExistsUpdated(config *v1beta1.NotificationConfiguration, external *awss3.GetBucketNotificationConfigurationResponse) bool {
+	return (config == nil || len(config.TopicConfigurations) == 0 || len(config.QueueConfigurations) == 0 || len(config.LambdaFunctionConfigurations) == 0) &&
+		len(external.QueueConfigurations) == 0 && len(external.LambdaFunctionConfigurations) == 0 && len(external.TopicConfigurations) == 0
+}
+
+func bucketStatus(config *v1beta1.NotificationConfiguration, external *awss3.GetBucketNotificationConfigurationResponse) ResourceStatus {
+	if notExistsUpdated(config, external) {
```

Comment:
The reason I separated it out because go-cyclo flagged it as being too high - I'm not really sure what's better of handling this is - but I'm open to ideas :)

Reply:
I believe you can use `// nolint:gocyclo` for such functions that consist of only `if` statements as long as they keep to be simple enough.